File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webapi/service/types/GetEndpointContext.java
package org.opcfoundation.webapi.service.types;

import org.eclipse.milo.opcua.stack.core.types.structured.RequestHeader;
import org.springframework.lang.Nullable;
import org.springframework.web.context.request.NativeWebRequest;

import java.util.List;

public class GetEndpointContext extends ServiceContext {
    final private String endpointUrl;
    final private List<String> localeIds;
    final private List<String> profileUris;

    public GetEndpointContext(
            @Nullable String endpointUrl,
            List<String> localeIds,
            List<String> profileUris,
            @Nullable String serverUri,
            @Nullable RequestHeader header,
            @Nullable NativeWebRequest webRequest)
    {
        super(serverUri,header,webRequest);
        this.endpointUrl = (null == endpointUrl) ? "" : endpointUrl;
        this.localeIds = localeIds;
        this.profileUris = profileUris;
    }

    public String getEndpointUrl() {
        return endpointUrl;
    }

    public List<String> getLocaleIds() {
        return localeIds;
    }

    public List<String> getProfileUris() {
        return profileUris;
    }
}

File: opcua-webservice-node/src/server/types/contexts/GetEndpointContext.ts
import { RequestHeader } from "opcua-webapi";
import { ServiceContext } from "./ServiceContext";

export class GetEndpointContext extends ServiceContext
{
    private _endpointUrl : string;
    private _localeIds : Array<string>;
    private _profileUris : Array<string>;

    constructor(
        endpointUrl?: string, 
        localeIds?: Array<string>, 
        profileUris?: Array<string>,
        serverUri?: string, 
        requestHeader? : RequestHeader)
    {
        super(serverUri, requestHeader);

        this._endpointUrl = (endpointUrl) ? endpointUrl : "";
        this._localeIds = (localeIds) ? localeIds : [];
        this._profileUris = (profileUris) ? profileUris : [];
    }

    get endpointUrl() : string
    {
        return this._endpointUrl;
    }

    get localeIds() : Array<string>
    {
        return this._localeIds;
    }

    get profileUris() : Array<string>
    {
        return this._profileUris;
    }
}